frontends/web/src/routes/device/bitbox02/createbackup.tsx

```diff
@@ -15,83 +15,67 @@
  * limitations under the License.
  */
 
-import { Component } from 'react';
-import { createBackup } from '../../../api/bitbox02';
-import { translate, TranslateProps } from '../../../decorators/translate';
+import { useState } from 'react';
+import { checkBackup, createBackup as createBackupAPI } from '../../../api/bitbox02';
 import { apiPost } from '../../../utils/request';
 import { alertUser } from '../../../components/alert/Alert';
 import { confirmation } from '../../../components/confirm/Confirm';
 import { Button } from '../../../components/forms';
 import { WaitDialog } from '../../../components/wait-dialog/wait-dialog';
+import { useTranslation } from 'react-i18next';
+import { Backup } from '../components/backup';
 
-interface CreateProps {
+type TProps = {
     deviceID: string;
 }
 
-type Props = CreateProps & TranslateProps;
+export const Create = ({ deviceID }: TProps) => {
+  const [creatingBackup, setCreatingBackup] = useState(false);
+  const [disabled, setDisabled] = useState(false);
+  const { t } = useTranslation();
 
-interface State {
-    creatingBackup: boolean;
-    disabled: boolean;
-}
-
-class Create extends Component<Props, State> {
-  public readonly state: State = {
-    creatingBackup: false,
-    disabled: false,
-  };
-
-  private createBackup = () => {
-    this.setState({ creatingBackup: true });
-    createBackup(this.props.deviceID)
-      .then(() => this.setState({ creatingBackup: false, disabled: false }))
+  const createBackup = () => {
+    setCreatingBackup(true);
+    createBackupAPI(deviceID)
+      .then(() => {
+        setCreatingBackup(false);
+        setDisabled(false);
+      })
       .catch(() => {
-        alertUser(this.props.t('backup.create.fail'));
+        setCreatingBackup(false);
+        setDisabled(false);
+        alertUser(t('backup.create.fail'));
       });
   };
 
-  private maybeCreateBackup = () => {
-    this.setState({ disabled: true });
-    apiPost('devices/bitbox02/' + this.props.deviceID + '/backups/check', {
-      silent: true,
-    }).then(check => {
-      if (check.success) {
-        confirmation(this.props.t('backup.create.alreadyExists'), result => {
-          if (result) {
-            this.createBackup();
-          } else {
-            this.setState({ disabled: false });
-          }
-        });
-        return;
-      }
-      this.createBackup();
-    });
+  const maybeCreateBackup = async () => {
+    setDisabled(true);
+    const check = await checkBackup(deviceID, true);
+    if (check) {
+      confirmation(t('backup.create.alreadyExists'), result => {
+        if (result) {
+          createBackup();
+        } else {
+          setDisabled(false);
+        }
+      });
+      return;
```

Comment:
Now using `checkBackup`. Just one question, I removed the last `createBackup()`. I wonder why we had it there? Seems like it doesn't have any effect..?